enterprise/server/ociregistry/ociregistry.go

```diff
@@ -327,33 +313,39 @@ func (r *registry) handleBlobsOrManifestsRequest(ctx context.Context, w http.Res
 		}
 	}
 	w.WriteHeader(upresp.StatusCode)
+	if inreq.Method == http.MethodHead {
+		return
+	}
 
-	contentLength, hasLength, err := parseContentLengthHeader(upresp.Header.Get(headerContentLength))
+	hasLength := upresp.Header.Get(headerContentLength) != ""
+	contentLength, err := strconv.ParseInt(upresp.Header.Get(headerContentLength), 10, 64)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("could not parse %s header (value '%s') from upstream: %s", headerContentLength, upresp.Header.Get(headerContentLength), err), http.StatusNotFound)
-		return
+		hasLength = false
 	}
 
 	hasContentType := upresp.Header.Get(headerContentType) != ""
 	contentType := upresp.Header.Get(headerContentType)
 
-	// TODO(dan): Docker Hub responds with Docker-Content-Digest headers. Other registries may not. Make code resilient to header absence.
-	hash, err := parseDockerContentDigestHeader(upresp.Header.Get(headerDockerContentDigest))
-	if err != nil {
-		http.Error(w, fmt.Sprintf("could not parse %s header (value '%s') from upstream: %s", headerDockerContentDigest, upresp.Header.Get(headerDockerContentDigest), err), http.StatusNotFound)
-		return
-	}
-
-	if upresp.StatusCode == http.StatusOK && inreq.Method == http.MethodGet && hasLength && hash != nil && hasContentType {
-		err := writeBlobOrManifestToCacheAndResponse(ctx, upresp.Body, w, bsClient, acClient, resolvedRef, ociResourceType, *hash, contentType, contentLength)
+	cacheable := upresp.StatusCode == http.StatusOK && hasLength && resolvedRefIsDigest && hasContentType
+	if !cacheable {
+		_, err = io.Copy(w, upresp.Body)
 		if err != nil && err != context.Canceled {
-			log.CtxWarningf(ctx, "error writing response body to cache for '%s': %s", inreq.URL, err)
+			log.CtxWarningf(ctx, "error writing response body for '%s': %s", inreq.URL, err)
 		}
-	} else {
+	}
```

Comment:
Is there a missing return here? do you want to proceed to the code below if cacheable is false?

Reply:
Yes. Fixed.